generador de archivos residencias ok

rochimgg opened the pull request.

Commit:
generador de archivos residencias

alelevinas committed:
Cambio el nombre de la carpeta por una sin espacios

alelevinas committed:
Merge branch 'master' into Matrimonios-estables

alelevinas committed:
Ejecuto el generador en travis con parametros por argumento

alelevinas committed:
Arreglo el nombre del .py en travis

alelevinas committed:
Cambio parametros de entrada

alelevinas committed:
Agrego cat salida.txt al travis

alelevinas committed:
Saco las notificaciones por email

mvbattan commented on .travis.yml:
```diff
@@ -5,3 +5,8 @@ install:
     - sudo apt-get install -y pylint
 script:
     - ls | grep \.py3$ | xargs
```
Acá debería correr pylint, después lo seteo

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+	return q_ordered
+	
+def arrayDisordering(a):
+	i=len(a)
```
`return Random.shuffle(a)` ?

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+		a[j],a[i] = a[i], a[j]
+	return
+	
+def preferenceDisordering(q):
```
Cuál es el objetivo de esta función ?

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+			f.write("\"\"\"HOSPITAL: "+str(j+1)+" ESPECIALIDAD:"+str(k+1)+"\"\"\" "+ str(meritCreator(n))+'\n')# M RENGLONES CON EL ORDEN DE MERITO DE CADA HOSP SEPARADO CON ESPACIOS
+
+
+file_name = sys.argv[1]   #input("Ingrese el nombre del archivo de salida: ")
```
Borraría el código comentado

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+N = sys.argv[2] #input("Ingrese la cantidad de estudiantes: ")
+M = sys.argv[3] #input("Ingrese la cantidad de hospitales: ")
+int(N)
+int(M)
```
Acá deberíamos castear a int, y atrapar la excepción correspondiente si no se puede. Así evitamos castear en las funciones auxiliares.

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+	for i in range(int(m)):
+		for j in range (l):
+			q_ordered.append(str(i+1)+','+str(j+1))
+			k=k+1
```
Necesitamos a `k` ?

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+		s=""
+		Q = orderedPreferenceMaker(m,l)
+		preferenceDisordering(Q)
+		for	i in range(int(m)):
```
Borraría espacios entre `for` e `i` de modo de ser consistente con el estilo

mvbattan replied:
Creo que puede resolverse con un  `join` de los elementos de la lista

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+	H=[]
+	s=""
+	for i in range(int(n)):
+		H.append(i+1)
```
`H = range(1, int(n))` ? Reemplazaría a `[]` y al `for`

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+	s=""
+	for i in range(int(m)):
+	 	q.append(random.randint(i,j))
+	 	s=s+" "+str(q[i])
```
Con `s = s + " " + str(random.randint(i, j))` nos ahorramos de crear a `q` para que ocupe lugar en el stack.

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+
+def randomPreferencePrinter(m,n,f): #IMPRIME m PREFERENCIAS DE HOSPITALES Y ESPECIALIDADES PARA n ALUMNOS
+	for i in range(int(n)):
+		f.write("\"\"\"ALUMNO: "+str(i+1)+"\"\"\" "+str(preferenceCreator(m,L))+'\n') # N RENGLONES CON LAS PREFERENCIAS DE LOS ESTUDIANTES SEPARADAS POR ESPACIOS
```
Mmm, creo que habría que cambiar el formato de salida según el pedido (no estaría haciendo falta escribir `ALUMNO` en el output).

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+def randomMeritPrinter(n,m,f):
+	for j in range(int(m)):
+		for k in range(L):
+			f.write("\"\"\"HOSPITAL: "+str(j+1)+" ESPECIALIDAD:"+str(k+1)+"\"\"\" "+ str(meritCreator(n))+'\n')# M RENGLONES CON EL ORDEN DE MERITO DE CADA HOSP SEPARADO CON ESPACIOS
```
Mismo comentario para `HOSPITAL`

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+		H.append(i+1)
+	arrayDisordering(H)
+	for i in range(int(n)):
+		s=s+" "+str(H[i])
```
Esto creo que también podría resolverse con un `join` de la lista.

mvbattan commented on stable_matching/generador_archivos_residencia.py:
```diff
@@ -0,0 +1,89 @@
+
+def orderedPreferenceMaker(m,l):
+	k=0	
+	s=""
```
Necesitamos a `s` ?

mvbattan reviewed (commented):
Lo voy a estar probando más tarde. Después seteo el linter cosa que nos corrija el espaciado / indentado. @Rochimgg Te dejo un par de comentarios

alelevinas committed:
Merge branch 'master' into Matrimonios-estables

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Matrimonios-estables

Commit:
ANDA =D

Chicos disculpen por tenerlos hasta tan tarde con esto, ya maniana voy a
aprovechar el dia para hacer un lindo informe. Que tengan una linda
noche!

Commit:
Merge branch 'Matrimonios-estables' of https://github.com/mvbattan/tda-tp0-2017-1q into Matrimonios-estables
